codegen/templates/http_client.tmpl

```diff
@@ -170,35 +177,39 @@ func (c *{{$clientName}}) {{title .Name}}(
 	switch res.StatusCode {
 		case {{.OKStatusCode.Code}}:
 			var responseBody {{unref .ResponseType}}
+			{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(&responseBody)
+			{{else -}}
+			err = res.ReadAndUnmarshalNonPointerValue(&responseBody)
```

Comment:
`ReadAndUnmarshalNonPointerValue` takes a pointer but its name means the opposite.

can we use `ReadAndUnmarshalNonPointerValue` to replace `ReadAndUnmarshalBody`?